network/server_test.go

```diff
@@ -690,19 +698,48 @@ func TestMinimumBootNodeCount(t *testing.T) {
 			shouldFail: false,
 		},
 	}
+
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			_, createErr := CreateServer(&CreateServerParams{
+			srv, createErr := CreateServer(&CreateServerParams{
 				ConfigCallback: func(c *Config) {
 					c.Chain.Bootnodes = tt.bootNodes
+					c.NoDiscover = false
 				},
 			})
-
 			if tt.shouldFail {
 				assert.Error(t, createErr)
 			} else {
 				assert.NoError(t, createErr)
+				assert.NoError(t, srv.Close())
 			}
 		})
 	}
 }
+
+func TestTemporaryDial(t *testing.T) {
+	defaultConfig := &CreateServerParams{
+		ConfigCallback: func(c *Config) {
+			c.NoDiscover = true
+		},
+	}
+	servers, err := createServers(2, map[int]*CreateServerParams{0: defaultConfig, 1: defaultConfig})
+
+	assert.NoError(t, err)
```

Comment:
This shouldn't be an assert, but a fatal test stopping condition

Reply:
Done,Commit-[link](https://github.com/0xPolygon/polygon-edge/pull/343/commits/ea9b338d7b7cced2866e0ee80785c2ccd4fdf7f2)